### PR TITLE
add node affinity

### PR DESCRIPTION
**Pull request by soulseen**
Signed-off-by: soulseen <sunzhu@yunify.com>

### CODE REVIEW

**Comment by soulseen** [comment]
The user does not have permission to modify the label name, because the label name is also asscoiate to the jenkins pipeline, and Kubesphere will take ci label as the official label to manage. User only can add ci label to node which needs to run.
@magicsong

**Comment by magicsong** [comment]
@soulseen at the view of decoupling, the labels of s2i should be different from labels of jenkins. As s2ioperator is an independent component and can be installed in any kubernetes while not all kubernetes have jenkins installed.

**Comment by runzexia** [comment]
Need to add default value, only receive user settings now?

File: pkg/controller/s2irun/ksbuilder_jobs.go [hunk 1]
```diff
@@ -250,6 +253,69 @@ func setJobLabelAnnotations(instance *devopsv1alpha1.S2iRun, config devopsv1alph
...
+		Operator: corev1.TolerationOpExists,
+		Effect:   corev1.TaintEffectPreferNoSchedule,
+	}
+
+	tolerations := []corev1.Toleration{
+		tolerationNoSchedule,
+		tolerationPreferNoSchedule,
+	}
+
+	nodeSelectorRequirements := corev1.NodeSelectorRequirement{
```
Comment (runzexia) [inline pkg/controller/s2irun/ksbuilder_jobs.go]: do not need set node selector

**Comment by soulseen** [comment]
still keep default value.
```	
        TaintKey          = "kubesphere.io/ci"
	NodeAffinityKey   = "kubesphere.io/worker"
	NodeAffinityValue = "ci"
```

File: config/samples/devops_v1alpha1_s2ibuilder.yaml [hunk 2]
```diff
@@ -15,4 +15,7 @@ spec:
     tag: latest
     builderPullPolicy: if-not-present
     revisionId: "staging"
+    nodeAffinityKey: "affinityKey1"
+    nodeAffinityValue: ["affinityValue1"]
```
[the remark this answers is not in the document]
Comment (soulseen) [reply config/samples/devops_v1alpha1_s2ibuilder.yaml]: done

**Comment by magicsong** [comment]
/lgtm

<sub>Sent with <a href="http://githawk.com">GitHawk</a></sub>

**Comment by magicsong** [comment]
/approve

<sub>Sent with <a href="http://githawk.com">GitHawk</a></sub>

**Pull request merged**